dash/App/Accounts/Add/Components/index.js

```diff
@@ -0,0 +1,278 @@
+      </div>
+      <div
+        data-testid='addHotAccountSecretSubmitButton'
+        className='addAccountItemOptionSubmit'
+        onMouseDown={handleSubmit}
+      >
+        Next
+      </div>
+    </div>
+  )
+}
+
+function PasswordInput({ getError, nextStep, title, buttonText }) {
+  const [error, setError] = useState(null)
+  const [ready, setReady] = useState(false)
+  const inputRef = useRef(null)
+
+  const debounceInput = useCallback(
+    debounce(() => {
+      const {
+        current: { value }
+      } = inputRef
+      const err = getError(value)
+      if (value) {
+        setError(err)
+        setReady(!err)
+      }
+    }, 300),
+    [debounce]
+  )
+
+  useEffect(() => {
+    inputRef.current.addEventListener('input', debounceInput)
+    //TODO: Do we need to do this, does react handle it?
+    // return () => {
+    //   inputRef.current.removeEventListener('input', debounceInput)
+    // }
+  })
+
+  const resetState = () => {
+    setReady(false)
+    setError('')
+  }
+
+  const handleSubmit = () => {
+    const {
+      current: { value }
+    } = inputRef
+    const passwordError = getError(value)
+    if (passwordError) return setError(passwordError)
+    resetState()
+    nextStep(value)
+  }
+
+  return (
+    <div style={{ textAlign: 'center', width: '100%' }} className='addAccountItemOptionSetupFrame'>
+      <div className='addAccountItemOptionTitle'>{title}</div>
+      {error && <div style={{ color: 'red' }}>{error}</div>}
+      <div className='addAccountItemOptionInputPhrase addAccountItemOptionInputPassword'>
+        <div className='addAccountItemOptionSubtitle'>password must be 12 characters or longer</div>
+        <form onSubmit={handleSubmit}>
+          <input
+            autoFocus
+            type='password'
+            tabIndex='-1'
+            ref={inputRef}
+            onChange={() => {
+              setError('')
+              setReady(false)
+            }}
+          />
+        </form>
+      </div>
+      {/* TODO: Maybe use CSS to make button clearly un-clickable rather than dissappearing? */}
+      {ready && (
+        <div className={'addAccountItemOptionSubmit'} onMouseDown={handleSubmit}>
+          {buttonText}
+        </div>
+      )}
+    </div>
+  )
+}
+
+function CreatePassword({ secret, newAccountType }) {
+  const title = 'Create Password'
+  const buttonText = 'Continue'
+
+  const getError = (password) => {
+    if (password.length < 12) return 'PASSWORD MUST BE AT LEAST 12 CHARACTERS LONG'
+    const {
+      feedback: { warning },
+      score
+    } = zxcvbn(password)
+    if (score > 2) return
+
+    return (warning || 'PLEASE ENTER A STRONGER PASSWORD').toUpperCase()
+  }
+
+  const nextStep = (password) =>
+    navForward(newAccountType, {
+      secret,
+      password
+    })
+
+  return <PasswordInput {...{ getError, secret, nextStep, title, buttonText }} />
+}
+
+function ConfirmPassword({ password, secret, newAccountType, createSignerMethod }) {
```

Comment:
it might be a little clearer to pass down a function like `onConfirm()` that navigates to wherever we need to go, that way this component, which is concerned with entering a password, doesn't need to know things about account types and signer methods and could theoretically be used anywhere